### ISSUE TITLE
@Listen decorator should accept constant expressions

### ISSUE THREAD

**Issue by benkeil** [ionic-team/stencil#2924]
**Stencil version:**
```
@stencil/core@2.5.2
```

**I'm submitting a:**

[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior**

`@Listen(WidgetSetupProvidedEvent.IDENTIFIER)` results in a build error `"invalid decorator argument: WidgetSetupProvidedEvent.IDENTIFIER`.

**Expected behavior**

It should compile.

**Side Note**

The `@Event` decorator already accepts constants.

```typescript
@Event({
  eventName: WidgetSetupEvent.IDENTIFIER,
  bubbles: true,
})
widgetSetupEventDetailsEventEmitter: EventEmitter<WidgetSetupEventDetails>;
```

**Comment by r3dDoX** [comment]
This would be very helpful to us for handling events throughout the application. A String literal is just not maintainable on a larger code base. I was wondering if there is a workaround with a rollup plugin or something that could at least at build time inline the constant as string literal?

**Comment by cary-smith** [comment]
This has been an issue for a long time unfortunately and may not be resolved soon.

https://github.com/ionic-team/stencil/issues/398

**Comment by splitinfinities** [comment]
Hey there, thank you for the patience getting back to you. The new team is getting started and we're working through the backlog now.

Hey there! I wanted to let you know that we're going to look into this. No timeframe, but we can see the use case for something like this. We're going to investigate this and circle back afterward.

**Comment by aschenoni** [comment]
Have there been any updates on this? Still a very relevant feature.

**Comment by rwaskiewicz** [comment]
This issue has been ingested into our backlog. We don't have a timeline for this at the moment - I know that this was attempted in the past and their were issues surrounding it, so some fact finding is going to  be required on our side.  I do believe this change would be a breaking change to the runtime in some scenarios, and is most likely to be slated for a future major version release of Stencil

**Comment by tanner-reits** [comment]
This was resolved by #4746 and released in [Stencil v4.3.0](https://github.com/ionic-team/stencil/releases/tag/v4.3.0)
